### PR TITLE
buildah: use the same logic for XDG_RUNTIME_DIR as podman

### PR DESCRIPTION
**Pull request by giuseppe**
if XDG_RUNTIME_DIR is not set, try in the order:

- /var/run/user/$UID/run
- $HOME/rundir

also set the XDG_RUNTIME_DIR so that the OCI runtime will
use the same settings.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: util/util.go [hunk 1]
```diff
@@ -472,11 +473,31 @@ func UnsharedRunrootPath(uid string) (string, error) {
 	if envRuntimeDir, haveRuntimeDir := os.LookupEnv("XDG_RUNTIME_DIR"); haveRuntimeDir {
 		return filepath.Join(envRuntimeDir, "run"), nil
 	}
+	var runtimeDir string
 	// If $XDG_RUNTIME_DIR is not defined, but we know our UID...
 	if uid != "" {
-		return filepath.Join("/var/run/user", uid, "run"), nil
+		tmpDir := filepath.Join("/var/run/user", uid, "run")
```
Comment (TomSweeneyRedHat) [inline util/util.go]: Podman doesn't seem to have "run" on the end, is this correct or did I misread the podman code?

**Comment by TomSweeneyRedHat** [comment]
Red Hat CI isn't happy.  I think, but don't know that it's a flake.

File: util/util.go (hunk 1 above)
Comment (giuseppe) [reply util/util.go]: I've left this part similar to the previous code in Buildah.  Should I drop the last "run"?

/cc @nalind

Comment (giuseppe) [reply util/util.go]: thinking more of it, it seems correct the version without the "run" so I've pushed a new version now.  Thanks for catching this issue!

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by vbatts** [comment]
tested and it works on centos7

**Comment by rhatdan** [comment]
@jlebon any idea what is going on here with papr?

**Comment by TomSweeneyRedHat** [comment]
LGTM, assuming happy tests.

**Comment by jlebon** [comment]
Infra is currently experiencing difficulties. Ops is investigating.

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by vrothberg** [comment]
Once https://github.com/containers/buildah/pull/1090 is merged, tests will be calm again.

**Comment by vrothberg** [comment]
@giuseppe, you can now rebase and test should be fine again

**Comment by giuseppe** [comment]
rebased :arrow_up:

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**